src/DataWrangling/EN4/EN4.jl

```diff
@@ -0,0 +1,257 @@
+
+    if variable_is_three_dimensional(metadata)
+        TZ = Bounded
+        LZ = Center
+        z = EN4_z
+        halo = (horizontal_halo..., 3)
+        sz = (Nx, Ny, Nz)
+    else # the variable is two-dimensional
+        TZ = Flat
+        LZ = Nothing
+        z = nothing
+        halo = horizontal_halo
+        sz = (Nx, Ny)
+    end
+
+    grid = LatitudeLongitudeGrid(architecture, Float32; halo, longitude, latitude, z,
+                                 size = sz,
+                                 topology = (TX, TY, TZ))
+
+    return Field{loc...}(grid)
+end
+
+# Only temperature and salinity need a thorough inpainting because of stability,
+# other variables can do with only a couple of passes. Sea ice variables 
+# cannot be inpainted because zeros in the data are physical, not missing values.
+function default_inpainting(metadata::EN4Metadata)
+    if metadata.name in [:temperature, :salinity]
+        return NearestNeighborInpainting(Inf)
+    elseif metadata.name in [:sea_ice_fraction, :sea_ice_thickness]
+        return nothing
+    else
+        return NearestNeighborInpainting(5)
+    end
+end
+
+"""
+    EN4_field(metadata::EN4Metadata;
+               architecture = CPU(),
+               inpainting = nothing,
+               mask = nothing,
+               horizontal_halo = (7, 7),
+               cache_inpainted_data = false)
+
+Return a `Field` on `architecture` described by `EN4Metadata` with
+`horizontal_halo` size.
+If not `nothing`, the `inpainting` method is used to fill the cells
+within the specified `mask`. `mask` is set to `EN4_mask` for non-nothing
+`inpainting`.
+"""
+function EN4_field(metadata::EN4Metadata;
+                    architecture = CPU(),
+                    inpainting = default_inpainting(metadata),
+                    mask = nothing,
+                    horizontal_halo = (7, 7),
+                    cache_inpainted_data = true)
+                    
+    field = empty_EN4_field(metadata; architecture, horizontal_halo)
+    inpainted_path = inpainted_metadata_path(metadata)
+
+    if !isnothing(inpainting) && isfile(inpainted_path)
+        file = jldopen(inpainted_path, "r")
+        maxiter = file["inpainting_maxiter"]
+
+        # read data if generated with the same inpainting
+        if maxiter == inpainting.maxiter
+            data = file["data"]
+            close(file)
+            copyto!(parent(field), data)
+            return field
+        end
+
+        close(file)
+    end
+
+    download_dataset(metadata)
+    path = metadata_path(metadata)
+    ds = Dataset(path)
+    shortname = short_name(metadata)
+
+    if variable_is_three_dimensional(metadata)
+        data = ds[shortname][:, :, :, 1]
+        data = reverse(data, dims=3)
+    else
+        data = ds[shortname][:, :, 1]
+    end        
+
+    close(ds)
+    
+    # Convert data from Union(FT, missing} to FT
+    FT = eltype(field)
+    data[ismissing.(data)] .= 1e10 # Artificially large number!
```

Comment:
```suggestion
    data[ismissing.(data)] .= NaN # Artificially large number!
```

Reply:
is this the issue?

Reply:
I think this is part of the issue; it may be that the Boolean mask for EN4 isn't being created properly, such that it is `false` everywhere - in this line: 

This would lead to the final field having a bunch of `missing` values, which are then set to `1e10` in the above.

Reply:
I think the inpainting will only replace NaN. If there are 1e10 in the file, those values will not be touched by inpainting. That can in turn lead to 1e10 values in the final field

Reply:
EN4 marks land as `missing`, not a junk value. The `1e10` is introduced somewhere in this code, I am trying to pin down where.

Reply:
I figured this out - the inpainting is working now; though I suspect there might be some land mask distortion...